conformance_tests/core/test_memory/src/test_memory_export_import.cpp

```diff
@@ -33,14 +33,224 @@ namespace lzt = level_zero_tests;
+    size_t size, uint8_t pattern, test_memory_type_t test_memory_type,
+    ze_external_memory_type_flag_t external_memory_type_flag,
+    void **exported_memory, int *fd, ze_image_handle_t *image_handle) {
+
+  auto result = ZE_RESULT_SUCCESS;
+
+  /* Export Memory As DMA_BUF*/
+  ze_external_memory_export_desc_t export_desc = {};
+  export_desc.stype = ZE_STRUCTURE_TYPE_EXTERNAL_MEMORY_EXPORT_DESC;
+  export_desc.flags = external_memory_type_flag;
+
+  if (test_memory_type != test_memory_type_t::TEST_MEMORY_TYPE_HOST) {
+    if (test_memory_type > test_memory_type_t::TEST_MEMORY_TYPE_DEVICE) {
+      ze_image_format_t format = {};
+      format.layout = ZE_IMAGE_FORMAT_LAYOUT_8;
+      format.type = ZE_IMAGE_FORMAT_TYPE_UINT;
+      format.x = ZE_IMAGE_FORMAT_SWIZZLE_0;
+      format.y = ZE_IMAGE_FORMAT_SWIZZLE_0;
+      format.z = ZE_IMAGE_FORMAT_SWIZZLE_0;
+      format.w = ZE_IMAGE_FORMAT_SWIZZLE_0;
+
+      ze_image_desc_t image_desc = {};
+      image_desc.stype = ZE_STRUCTURE_TYPE_IMAGE_DESC;
+      image_desc.format = format;
+
+      switch (test_memory_type) {
+      case test_memory_type_t::TEST_MEMORY_TYPE_IMAGE_1D:
+        image_desc.type = ZE_IMAGE_TYPE_1D;
+        image_desc.width = 1024;
+        break;
+      case test_memory_type_t::TEST_MEMORY_TYPE_IMAGE_2D:
+        image_desc.type = ZE_IMAGE_TYPE_2D;
+        image_desc.width = 128;
+        image_desc.height = 8;
+        break;
+      case test_memory_type_t::TEST_MEMORY_TYPE_IMAGE_3D:
+        image_desc.type = ZE_IMAGE_TYPE_3D;
+        image_desc.width = 32;
+        image_desc.height = 8;
+        image_desc.depth = 4;
+        break;
+      default:
+        image_desc.type = ZE_IMAGE_TYPE_BUFFER;
+        image_desc.width = 1024;
+        break;
+      }
+
+      *image_handle = lzt::create_ze_image(context, device, image_desc);
+    }
+
+    *exported_memory = lzt::allocate_device_memory(size, 1, 0, &export_desc, 0,
+                                                   device, context);
+  } else {
+    ze_host_mem_alloc_desc_t host_alloc_desc = {};
+    host_alloc_desc.stype = ZE_STRUCTURE_TYPE_HOST_MEM_ALLOC_DESC;
+    host_alloc_desc.pNext = &export_desc;
+    *exported_memory =
+        lzt::allocate_host_memory(size, 1, 0, &export_desc, context);
+  }
+
+  // Fill the allocated memory with some pattern so we can verify
+  // it was exported successfully
+  auto cmd_bundle = lzt::create_command_bundle(
+      context, device, 0, ZE_COMMAND_QUEUE_MODE_DEFAULT,
+      ZE_COMMAND_QUEUE_PRIORITY_NORMAL, 0, 0, 0, is_immediate);
+
+  lzt::append_memory_fill(cmd_bundle.list, *exported_memory, &pattern,
+                          sizeof(pattern), size, nullptr);
+  if (test_memory_type > test_memory_type_t::TEST_MEMORY_TYPE_DEVICE) {
+    lzt::append_barrier(cmd_bundle.list);
+    lzt::append_image_copy_from_mem(cmd_bundle.list, *image_handle,
+                                    *exported_memory, nullptr);
+  }
+  lzt::close_command_list(cmd_bundle.list);
```

Comment:
you dont have to close immediate lists
```suggestion
  if (is_immediate) {
    lzt::close_command_list(cmd_bundle.list);
  }
```

Reply:
implemented